apps/sms/test/unit/compose_test.js

```diff
@@ -1454,156 +1492,173 @@ suite('compose_test.js', function() {
-            'size is changed after removing attachment');
-        });
+      test('removes the original attachment', function() {
+        sinon.assert.calledWith(
+          window.URL.revokeObjectURL,
+          this.attachment.mNextRender.dataset.thumbnail
+        );
+        assert.isFalse(document.body.contains(this.attachment.mNextRender));
       });
-
-      suite('cancel', function() {
-        setup(function() {
-          // trigger click on close
-          document.getElementById('attachment-options-cancel').click();
-        });
-        test('closes the menu', function() {
-          assert.isTrue(AttachmentMenu.close.called);
-        });
+      test('recalculates size', function() {
+        assert.equal(Compose.size, 0, 'size should be 0 after remove');
+        assert.notEqual(Compose.size, this.attachmentSize,
+          'size is changed after removing attachment');
       });
+    });
 
-      suite('replace', function() {
-        var request;
+    suite('replace', function() {
+      var requestPromise;
 
-        setup(function() {
-          this.replacement = mockImgAttachment(true);
-          request = {};
-          this.sinon.stub(Compose, 'requestAttachment').returns(request);
-          this.sinon.stub(Utils, 'getResizedImgBlob');
+      setup(function() {
+        this.replacement = mockImgAttachment(true);
+        this.sinon.stub(Compose, 'requestAttachment');
+        this.sinon.stub(Utils, 'getResizedImgBlob');
+      });
 
+      suite('onsuccess', function() {
+        setup(function() {
+          requestPromise = Promise.resolve(this.replacement);
+          Compose.requestAttachment.returns(requestPromise);
           // trigger click on replace
-          document.getElementById('attachment-options-replace').click();
+          call.items[2].method();
         });
 
-        suite('onsuccess', function() {
-          setup(function() {
-            request.onsuccess(this.replacement);
-          });
-
-          test('clicking on replace requests an attachment', function() {
-            assert.isTrue(Compose.requestAttachment.called);
-          });
-          test('removes the original attachment', function() {
+        test('clicking on replace requests an attachment', function() {
+          assert.isTrue(Compose.requestAttachment.called);
+        });
+        test('removes the original attachment', function(done) {
+          requestPromise.then(() => {
             sinon.assert.calledWith(
               window.URL.revokeObjectURL,
               this.attachment.mNextRender.dataset.thumbnail
             );
             assert.isFalse(document.body.contains(this.attachment.mNextRender));
-          });
-          test('inserts the new attachment', function() {
+          }).then(done, done);
+        });
+        test('inserts the new attachment', function(done) {
+          requestPromise.then(() => {
             assert.ok(this.replacement.mNextRender.parentNode);
```

Comment:
nit: maybe ``assert.isTrue(document.body.contains(this.attachment.mNextRender));```